index.js

```diff
@@ -1112,26 +1112,27 @@ class Watcher extends ReadyResource {
     return this.close()
   }
 
-  async _closeSnapshot (snap, opts) {
-    const snapshot = this[snap]
+  async _closeSnapshot (snapshot, mapped, opts) {
+    const name = snapshot === this.current ? 'current' : 'previous'
 
     if (snapshot) {
-      this[snap] = null
+      this[name] = null
```

Comment:
I passed down the actual properties into the method, but still did the reflective thing here

Otherwise here I would have to do:
```js
if (isCurrent) this.current = null
else this.previous = null
```

And same below for `this.mapped.*`

It's just simpler to have the property name defined and use it